fireant/queries/builder/dataset_blender_query_builder.py

```diff
@@ -58,56 +61,95 @@ def _find_dataset_fields_needed_to_be_mapped(dataset):
+    """
+    Returns a list of tuples shaped as blended dataset field and its matching field in the provided fields. Blender
+    fields that have to match in the dataset fields will be ignored. Please note that the Field class has a custom
+    dunder hash method.
+
+    :param dataset_fields: A set of Field instances pertaining to a single dataset (i.e. primary or secondary).
+    :param blender_dataset_fields: A list of Field instances in the blender dataset.
+    :param dimension_map: A dict mapping Field instances from different datasets.
+                          Often used to express the equivalency of primary and secondary dataset fields.
+    :return: A list of tuples shaped as blended dataset field and its matching field in the provided fields.
+    """
+    for field in blender_dataset_fields:
         if dimension_map is not None and field.definition in dimension_map:
             yield field, dimension_map[field.definition]
-        if field.definition in dataset.fields:
+        if field.definition in dataset_fields:
             yield field, field.definition
             # also yield the dataset field mapped to itself so that any reference to this field (from blender or
             # from dataset) can be used.
             yield field.definition, field.definition
-        if field in dataset.fields:
+        if field in dataset_fields:
             yield field, field
 
 
-def _datasets_and_field_maps(blender):
+def _datasets_and_field_maps(blender_dataset, filters):
+    """
+    Returns a list of tuples shaped as dataset and its fields mapped per the blender dataset ones. The field maps
+    are later used for knowing which columns to query on the respective primary and secondary datasets, given the
+    columns selected for the blender dataset.
+
+    :param blender: A DataSetBlender instance.
+    :param filters: A list of Filter instances, present on the blender query. This is used for creating dimensions
+                    derived from filters, such as the set dimension.
+    :return: A list of tuples shaped as dataset and its fields mapped per the blender dataset.
+    """
     from fireant.dataset.data_blending import DataSetBlender
 
     def _flatten_blend_datasets(dataset) -> List:
-        primary = dataset.primary_dataset
-        secondary = dataset.secondary_dataset
-        dataset_fields = _find_dataset_fields_needed_to_be_mapped(dataset)
+        primary_dataset = dataset.primary_dataset
+        secondary_dataset = dataset.secondary_dataset
+        blender_dataset_fields = apply_set_dimensions(
+            _find_dataset_fields_needed_to_be_mapped(dataset), filters, dataset,
+        )
+
+        primary_dataset_fields = set(apply_set_dimensions(primary_dataset.fields, filters, primary_dataset))
+        secondary_dataset_fields = set(apply_set_dimensions(secondary_dataset.fields, filters, secondary_dataset))
+
+        primary_dimension_per_blender_dimension = dict(_map_field(
```

Comment:
Not sure if I prefer this name over `blender_to_primary_field_map`, I find it a bit confusion. Also, does it only contain dimensions?

Reply:
Not only. Haha, I was renaming it as you commented on this.

Reply:
I've reverted that.